src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponentResult.cs

```diff
@@ -47,17 +37,32 @@ public class ViewComponentResult : ActionResult
         public Type ViewComponentType { get; set; }
 
         /// <summary>
+        /// Get the view data model.
+        /// </summary>
+        public object Model => ViewData?.Model;
+
+        /// <summary>
         /// Gets or sets the <see cref="ViewDataDictionary"/> for this result.
         /// </summary>
         public ViewDataDictionary ViewData { get; set; }
 
         /// <summary>
+        /// Gets or sets the <see cref="ITempDataDictionary"/> for this result.
+        /// </summary>
+        public ITempDataDictionary TempData { get; set; }
```

Comment:
@cjqian I recommend leaving the order alone since it muddles the PR. If you want to make things :100:% consistent, do it elsewhere -- perhaps in a tiny change that doesn't get reviewed (once you've done a few PRs).